### ISSUE TITLE
Add the ability to use extensions without hosts file

### ISSUE THREAD

**Issue by dennisvandehoef** [StevenBlack/hosts#2328]
I really like this project, and where the way it works. Though, I now found a limitation.

Within my pi hole setup, I would like to specify groups and for that I need separate lists.
Now with extensions I can (for example) use the gambling list and add it to my children's config, whereas I use the fake news list and add it to all.
Sadly, this would result in a lot of unnecessary duplicates, as all extensions include the base hosts file.

I came across https://github.com/StevenBlack/hosts/issues/358, and it seems there is a nice solution in there.
I can easily use all lists that are currently used to build the extensions. And at first it looks like a great idea, though it has a downside.

The strength of this project is the community, and the external lists it uses. If now a new external list will be added to the ones used for the extensions, I will miss out.

Therefore, It would be really great if we can also offer extension only lists.
If you agree, I would be happy to work on this myself and create a pull request.

**Comment by StevenBlack** [comment]
Thank you for this and your kind words Dennis @dennisvandehoef.

I think this is a good idea.

One issue though: the path `alternates/porn/hosts` is used by the amalgamated list, plus porn.

So where in the hierarchy would we store the porn-only list?

I'm open to suggestions on that.

**Comment by dennisvandehoef** [comment]
I am asking myself what the usage of these lists are besides within Pi Hole, and if it will be a problem for other methods of using.

I would suggest `alternates/porn/no_hosts` or (more beautiful, but less clear) `alternates/porn/limited`.

For the use cases that I can see, calling the file other than `hosts` should not cause issues. But the might be use cases that I did not think about.

- Pi Hole
  - The name of the file does not matter, only the content
- The HostMan example from the README
  - The screenshots seem to be able to use different filenames 
- (Manual / using pre-compiled versions) Hosts file on a computer
  - I am not expecting these people to use these files, though I would still not exclude them from my thoughts
  - The file name matters, but depending on how people currently download the file they also have methods of using this
    - Using Wget in a cron, save the file as `hosts`
    - Using Git, these probably already symlink or copy the files over after a pull, renaming possible
- (Manual / Self building) Hosts file on a computer
  - We probably be able to add this as command line options, if not already possible with the current
  - Basically the solution in https://github.com/StevenBlack/hosts/issues/358 would apply here

**Comment by dennisvandehoef** [comment]
I just scanned the issues, and notice that I created a duplicate of the following 2 issues.

- https://github.com/StevenBlack/hosts/issues/1429
- https://github.com/StevenBlack/hosts/issues/1954

I don't know why I did not find them in my first search, though linking them here to keep track of the total demand of the feature (and text that might be helpful in determining the specs of the implemented solution).

**Comment by StevenBlack** [comment]
This is done!  Closing.
